src/test/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprintTest.java

```diff
@@ -25,9 +25,340 @@
+                tmpBitArray = tmpFingerprintRepresentation.getBitArray();
+                int tmpLength = java.util.Arrays.toString(tmpBitArray).length();
+                System.out.println("\t\tNumber of positive bits " + tmpSeparateList.get(0) + ": " + tmpBitFingerprint.cardinality());
+                System.out.println("\t\tIndices of positive bits " + tmpSeparateList.get(0) + ": " + tmpBitFingerprint.asBitSet().toString());
+                System.out.println("\t\tCount for the bin with the index 5 " + tmpSeparateList.get(0) + ": " + tmpCountFingerprint.getCountForHash(5));
+                tmpResultPrintWriter.println(java.util.Arrays.toString(tmpBitArray).substring(1, tmpLength - 1));
+                tmpResultPrintWriter.flush();
+            } catch (NullPointerException | IllegalArgumentException  anException) {
+                System.out.println("Generation of one fingerprint did not work.");
+            }
+            // add all molecule maps
+            FragmentFingerprintTest.moleculeFragmentList.add(tmpMoleculeFragmentsMap);
+        }
+        // Objects necessary for the test are created (used only in @Test)
+        FragmentFingerprintTest.fragmentFingerprinter = new FragmentFingerprinter(FragmentFingerprintTest.fragmentList);
+        FragmentFingerprintTest.countFingerprintTest = FragmentFingerprintTest.fragmentFingerprinter.getCountFingerprint(FragmentFingerprintTest.moleculeFragmentList.get(FragmentFingerprintTest.moleculeFragmentList.size() - 1));
+        FragmentFingerprintTest.bitFingerprintTest = FragmentFingerprintTest.fragmentFingerprinter.getBitFingerprint(FragmentFingerprintTest.dataForGenerateBitFingerprint);
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Test methods" defaultstate="collapsed">
+    /**
+     * Tests the getBitFingerprint() method
+     * Test molecule: Variamycin
+     *
+     * @throws Exception if anything goes wrong
+     */
+    @Test
+   public void bitFingerprintTest() {
+        //Test number of positive indices
+        int tmpNumberPositiveBitsTest = 9;
+        Assertions.assertEquals(tmpNumberPositiveBitsTest, FragmentFingerprintTest.bitFingerprintTest.cardinality());
+        // Test size
+        Assertions.assertEquals(64,FragmentFingerprintTest.bitFingerprintTest.size());
+        //Test asBitSet
+        BitSet tmpBitSetTest = new BitSet();
+        tmpBitSetTest.set(3);
+        tmpBitSetTest.set(5);
+        tmpBitSetTest.set(9);
+        tmpBitSetTest.set(14);
+        tmpBitSetTest.set(16);
+        tmpBitSetTest.set(17);
+        tmpBitSetTest.set(18);
+        tmpBitSetTest.set(26);
+        tmpBitSetTest.set(27);
+        Assertions.assertEquals(tmpBitSetTest,FragmentFingerprintTest.bitFingerprintTest.asBitSet());
+        // Test getSetbits
+        int[] tmpArrayBitSetTest = {3,5,9,14,16,17,18,26,27};
+        Assertions.assertArrayEquals(tmpArrayBitSetTest, FragmentFingerprintTest.bitFingerprintTest.getSetbits());
+
```

Comment:
Maybe I'm dumb, but do you test different methods in one test? Is that necessary or could you split it into one test for each method. Tests are best when they are as atomic as possible.